waiter/src/waiter/service_description.clj

```diff
@@ -854,7 +854,7 @@
      If after the merge a permitted-user is not available, then `username` becomes the permitted-user."
     [{:keys [defaults headers service-description-template source-tokens token-authentication-disabled token-preauthorized]}
      waiter-headers passthrough-headers kv-store service-id-prefix username metric-group-mappings
-     service-description-builder assoc-run-as-user-approved?]
+     service-description-builder assoc-run-as-user-approved? throw-if-invalid?]
```

Comment:
Is a better approach to return `nil` when a service description would be invalid, and change the existing callers to throw on `nil`?

Reply:
We still need metadata on why the service description is invalid to report in our error messages.

Reply:
It could still return a map with partial results instead of nil.  I guess my larger point is that when it becomes desirable to have a function not throw in some cases, it seems like we should make it not throw in all cases, and rework the callers that expected an exception, as opposed to passing a parameter into the function causing it to throw in certain circumstances.  That seems cleaner to me.